File: 백준/Silver/1406. 에디터/에디터.java


import java.util.*;
import java.io.*;

public class Main {
    static Stack<String> left = new Stack<>();
    static Stack<String> right = new Stack<>();
    static int length;
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        String[] tmp = br.readLine().split("");
        for (String str : tmp) {
            left.push(str);
        }
        length = Integer.parseInt(br.readLine());

        for (int i = 0; i < length; i++) {
            String input = br.readLine();
            switch(input){
                case "L" :
                    if(!left.isEmpty()) right.push(left.pop()); break;
                case "D" :
                    if(!right.isEmpty()) left.push(right.pop()); break;
                case "B" :
                    if(!left.isEmpty()) left.pop(); break;
                default:
                    left.push(String.valueOf(input.charAt(2)));
            }
        }

        while (!left.isEmpty()) {
            right.push(left.pop());
        }
        while(!right.isEmpty()){
            sb.append(right.pop());
        }
        System.out.println(sb);
    }
}

File: 백준/Silver/1406. 에디터/에디터.py
import sys

left = list()
right = list()

d = {
    "L": lambda: right.append(left.pop()) if left else None,
    "D": lambda: left.append(right.pop()) if right else None,
    "B": lambda: left.pop() if left else None,
    "P": lambda x: left.append(x)
}

left = list(sys.stdin.readline().strip())

n = int(sys.stdin.readline().strip())

for _ in range(n):
    tmp = sys.stdin.readline().split()
    op = tmp[0]
    if op == "P":
        d[op](tmp[1])
    else:
        d[op]()

while right:
    d["D"]()
print("".join(left))
